### PR TITLE
Upgrade istio dependencies to 1.18.1

### PR DESCRIPTION
**Pull request by zhaohuabing**
* Change the Istio version from 1.18.2 to 1.18.1 since 1.18.2 envoy commit doesn't exist
* Upgrade all the Istio dependencies to 1.18.1

### CODE REVIEW

File: demo/metaprotocol-brpc/brpc-sample.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ spec:
     metadata:
       annotations:
         sidecar.istio.io/bootstrapOverride: aeraki-bootstrap-config
-        sidecar.istio.io/proxyImage: ghcr.io/aeraki-mesh/meta-protocol-proxy-debug:1.4.0
+        sidecar.istio.io/proxyImage: ghcr.io/aeraki-mesh/meta-protocol-proxy-debug:1.4.1
```
Comment (tanjunchen) [inline demo/metaprotocol-brpc/brpc-sample.yaml:31]: To be honest, I personally feel that the  image ghcr.io/aeraki-mesh/meta-protocol-proxy:1.4.1 is more suitable here, the debug name is misleading, and the image size is much larger than it.

File: .github/workflows/e2e-metaprotocol-gateway.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ on:
     - "**/*.md"
 
 env:
-  ISTIO_VERSION: 1.18.2
+  ISTIO_VERSION: 1.18.1
```
Comment (tanjunchen) [inline .github/workflows/e2e-metaprotocol-gateway.yaml:21]: I'm curious about the reason for downgrading the version.

Comment (tanjunchen) [reply .github/workflows/e2e-metaprotocol-gateway.yaml:21]: btw, There are so many branches here

image

File: demo/metaprotocol-brpc/brpc-sample.yaml (hunk 1 above)
Comment (zhaohuabing) [reply demo/metaprotocol-brpc/brpc-sample.yaml:31]: Good point, let's change to then non-debug image then.

File: .github/workflows/e2e-metaprotocol-gateway.yaml (hunk 2 above)
Comment (zhaohuabing) [reply .github/workflows/e2e-metaprotocol-gateway.yaml:21]: > ut the reason for downgrading the version.

The envoy commit for Istio 1.8.2 doesn't exist for some reason.  So I decided to use 1.8.1, we can upgrade to a later release when they come out.

https://github.com/istio/proxy/blob/3c27a1b0cf381ca854ccc3a2034e88c206928da2/WORKSPACE#L38

```bash
(base) ➜  envoy git:(main) g checkout f826f1dbddd3326abee19009ea9f019e8a9d5997    
fatal: reference is not a tree: f826f1dbddd3326abee19009ea9f019e8a9d5997
```

Comment (zhaohuabing) [reply .github/workflows/e2e-metaprotocol-gateway.yaml:21]: > btw, There are so many branches here
> 
> image

Those are development branches and can be deleted.

File: istio-1.18.2/LICENSE [hunk 3]
```diff
@@ -0,0 +1,202 @@
+
+                                Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
```
Comment (tanjunchen) [inline istio-1.18.2/LICENSE:5]: Does istio-1.18.2 directory need to be submitted?

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v1
         with:
-          go-version: 1.19
+          go-version: '1.20'
```
Comment (tanjunchen) [inline .github/workflows/ci.yaml:39]: It is recommended to merge the commit, there are too many  commit id.

Comment (zhaohuabing) [reply .github/workflows/ci.yaml:39]: All the commits will be squashed into one when this PR is merged into the master.

**Review by tanjunchen — APPROVED** [review]
good Job!

**Pull request merged**